frontend/src/components/notifications/ui-notification-boundary.tsx

```diff
@@ -89,27 +91,43 @@ export const UiNotificationBoundary: React.FC<PropsWithChildren> = ({ children }
     [dispatchUiNotification, t]
   )
 
-  const dismissNotification = useCallback((notificationUuid: string): void => {
-    setUiNotifications((old) => {
-      const found = old.find((notification) => notification.uuid === notificationUuid)
-      if (found === undefined) {
-        return old
+  const dismissNotification = useCallback(
+    (notificationUuid: string): void => {
+      const entry = getUiNotification(notificationUuid)
+      if (!entry) {
+        return
       }
-      return old.filter((value) => value.uuid !== notificationUuid).concat({ ...found, dismissed: true })
-    })
-  }, [])
+      setUiNotification(notificationUuid, { ...entry, dismissed: true })
+    },
+    [setUiNotification, getUiNotification]
+  )
+
+  const pruneNotification = useCallback(
+    (notificationUuid: string): void => {
+      if (!uiNotifications[notificationUuid]) {
+        return
+      }
+      removeUiNotification(notificationUuid)
+    },
+    [uiNotifications, removeUiNotification]
+  )
 
   const context = useMemo(() => {
     return {
       dispatchUiNotification: dispatchUiNotification,
       showErrorNotification: showErrorNotification,
-      dismissNotification: dismissNotification
+      dismissNotification: dismissNotification,
+      pruneNotification: pruneNotification
     }
-  }, [dismissNotification, dispatchUiNotification, showErrorNotification])
+  }, [dismissNotification, dispatchUiNotification, showErrorNotification, pruneNotification])
+
+  const allNotifications = useMemo(() => {
+    return Object.values(uiNotifications)
```

Comment:
i'm not sure how order-safe `Object.values` is. Please sort the notifications by date.